test/functional/neutronless/conftest.py

```diff
@@ -12,29 +12,190 @@
 #   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
+"""In addition to pytest fixtures this module also contains general classes.
 
-import json
-import os
-import pytest
-import re
-import traceback
-import sys
+    These classes are TestConfig, and Resource4TestTracker.
 
+    TestConfig:  This is a Python class designed to contain the configurable
+state and methods generally needed during testing.  The general approach is
+to provide a configuration specific to some test needs via the TestConfig
+objects methods (primarily, __init__, but also others e.g. "load_esd").
+
+   The expected caller of the TestConfig object is an Experiment fixture, where
+the experiment instantiates, and configures, a particular TestConfig instance, 
+before handing off to the test code.   Examples of "Experiments" can be found
+in esd/conftest.py.
+"""
+
+import collections
```

Comment:
Community-driven requirement would be:
```python
import collections
import json
import os
import pytest
import re
import sys
import traceback

from collections import deque
from collections import namedtuple
from copy import deepcopy
from inspect import currentframe as cf
from inspect import getframeinfo as gfi
from inspect import getouterframes as gof
```

Technically, though, the hacking guidelines would have you import at the module level and not at the function/method level... this is longstanding code, though... so I'm not sure if it's appropriate to fix it here... or not.